src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -647,20 +647,61 @@ private static TypeDeclDesc getTheNearestClass(String outerTypeDeclName,
      * @param secondTypeDecl second input type declaration
      * @return difference between type declaration name matching count
      */
-    private static int getTypeDeclarationNameMatchingCountDiff(String outerTypeDeclName,
+    private static int getAnonTypeDeclarationNameMatchingCountDiff(String outerTypeDeclName,
                                                                TypeDeclDesc firstTypeDecl,
                                                                TypeDeclDesc secondTypeDecl) {
         int diff = Integer.compare(
-            CheckUtil.typeDeclarationNameMatchingCount(
+            getAnonSuperTypeMatchingCount(
                 outerTypeDeclName, secondTypeDecl.getQualifiedName()),
-            CheckUtil.typeDeclarationNameMatchingCount(
+            getAnonSuperTypeMatchingCount(
                 outerTypeDeclName, firstTypeDecl.getQualifiedName()));
         if (diff == 0) {
             diff = Integer.compare(firstTypeDecl.getDepth(), secondTypeDecl.getDepth());
         }
         return diff;
     }
 
+    /**
+     * Calculates and returns the type declaration matching count when
+     * {@code typeDeclarationToBeMatched} is considered to be the superclass of an anonymous inner
+     * class.
```

Comment:
can it be ?
```
     * Calculates the type declaration matching count for the superclass of an anonymous inner
     * class.
```

Reply:
[Addressed]( https://github.com/checkstyle/checkstyle/compare/cf6d56c9d7895e7dc3b9fbf1eebd377cbca20d39..29c8ef5a1215145f06875b79bb75fd1830f765f2)